From f860aaa4f476aea4f7cfbd50f8965f7981eb1b75 Mon Sep 17 00:00:00 2001
From: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
Date: Sat, 10 Aug 2024 14:43:33 +0530
Subject: [PATCH] soc: qcom: qcom-geni-se: use memcpy_toio to copy into IO
 region

To access the IO space, memcpy_toio should be used.

Change-Id: Id58626e7df4d7847f241cb2e78b9e57559e34a84
Signed-off-by: Kathiravan Thirumoorthy <quic_kathirav@quicinc.com>
(cherry picked from commit bc3c5d2b25505feceb1de8b8b93e613cbc2ab808)
---
 drivers/soc/qcom/qcom-geni-se.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/soc/qcom/qcom-geni-se.c b/drivers/soc/qcom/qcom-geni-se.c
index 677b0cd45c5d..5fd89b7302ea 100644
--- a/drivers/soc/qcom/qcom-geni-se.c
+++ b/drivers/soc/qcom/qcom-geni-se.c
@@ -1367,7 +1367,7 @@ void geni_se_fw_load(struct geni_se *se, uint8_t se_mode)
 		cfg_val_off = *((uint16_t*)&hdr->firmware_offset);
 		fw32 = (uint32_t*)&fw[cfg_val_off];
 		cfg_limit = *((uint16_t*)&hdr->firmware_sz_in_items);
-		memcpy((void*)(se->base + SE_GENI_CFG_RAMn_OFFSET),
+		memcpy_toio((void*)(se->base + SE_GENI_CFG_RAMn_OFFSET),
 				fw32, (cfg_limit * sizeof(uint32_t)));
 	}
 
-- 
2.34.1

